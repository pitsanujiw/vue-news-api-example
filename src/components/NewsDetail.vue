<template>
  <div class="text-center">
    <v-dialog v-model="dialog" width="800">
      <template v-slot:activator="{ on, attrs }">
        <v-btn text v-bind="attrs" v-on="on" @click="onOpenDialog">
          See more
        </v-btn>
      </template>

      <v-card class="mx-auto">
        <v-img
          class="white--text align-end"
          height="200px"
          :src="newsArticle.urlToImage"
          gradient="to top right, rgba(100,115,201,.33), rgba(25,32,72,.7)"
        >
          <v-card-title>{{ newsArticle.title }}</v-card-title>
        </v-img>

        <v-card-subtitle class="pb-0 mt-5">{{
          onChangeFormatDate(newsArticle.publishedAt)
        }}</v-card-subtitle>
        <v-card-subtitle
          ><v-chip class="mt-2">{{
            newsArticle.source.name
          }}</v-chip></v-card-subtitle
        >

        <v-card-text class="text--primary">
          <div class="mb-2">{{ newsArticle.author || "-" }}</div>
          <v-divider></v-divider>
          <div class="mt-2">{{ newsArticle.content }}</div>
        </v-card-text>

        <v-card-actions>
          <v-btn color="dark" text :href="newsArticle.url" target="_blank">
            Explore
          </v-btn>
        </v-card-actions>
      </v-card>
    </v-dialog>
  </div>
</template>

<script>
import { mapActions } from 'vuex';

import { onChangeFormatDate } from '@/utils';

export default {
  name: 'NewsDetail',
  props: {
    newsArticle: {
      type: Object,
    },
  },
  data: () => ({
    dialog: false,
  }),
  methods: {
    ...mapActions(['addHistory']),
    onOpenDialog() {
      this.addHistory(this.newsArticle);
    },
    onChangeFormatDate,
  },
  computed: {},
};
</script>

<style>
</style>
