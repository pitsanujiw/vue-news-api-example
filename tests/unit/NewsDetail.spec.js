import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Vuetify from 'vuetify';

import NewsDetail from '../../src/components/NewsDetail.vue';

const vuetify = new Vuetify();
const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Vuetify);

describe('NewsCard', () => {
  const modules = {
    getters: {},
  };
  let store;

  beforeEach(() => {
    jest.clearAllMocks();
    store = new Vuex.Store({
      modules: {
        newModules: modules,
      },
    });
  });

  it('Renders NewsDetail Component', () => {
    const wrapper = shallowMount(NewsDetail, {
      store,
      localVue,
      vuetify,
      propsData: {
        newsArticle: {
          urlToImage: '__URL__',
          title: '__TITLE__',
          author: '__AUTHOR__',
          description: '__DESCRIPTION__',
          name: '__NAME__',
          source: {
            name: '__SOURCE_NAME__',
          },
        },
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
