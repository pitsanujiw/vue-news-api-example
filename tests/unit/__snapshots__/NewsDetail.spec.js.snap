// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsCard Renders NewsDetail Component 1`] = `
<div class="text-center">
  <v-dialog-stub opendelay="0" closedelay="0" contentclass="" origin="center center" retainfocus="true" transition="dialog-transition" width="800">
    <v-card-stub loaderheight="4" tag="div" class="mx-auto">
      <v-img-stub height="200px" gradient="to top right, rgba(100,115,201,.33), rgba(25,32,72,.7)" options="[object Object]" position="center center" src="__URL__" transition="fade-transition" class="white--text align-end">
        <v-card-title-stub tag="div">__TITLE__</v-card-title-stub>
      </v-img-stub>
      <v-card-subtitle-stub tag="div" class="pb-0 mt-5">01 January 2022</v-card-subtitle-stub>
      <v-card-subtitle-stub tag="div">
        <v-chip-stub activeclass="" tag="span" active="true" closeicon="$delete" closelabel="$vuetify.close" filtericon="$complete" class="mt-2">__SOURCE_NAME__</v-chip-stub>
      </v-card-subtitle-stub>
      <v-card-text-stub tag="div" class="text--primary">
        <div class="mb-2">__AUTHOR__</div>
        <v-divider-stub></v-divider-stub>
        <div class="mt-2"></div>
      </v-card-text-stub>
      <v-card-actions-stub tag="div">
        <v-btn-stub color="dark" tag="button" activeclass="" target="_blank" text="true" type="button">
          Explore
        </v-btn-stub>
      </v-card-actions-stub>
    </v-card-stub>
  </v-dialog-stub>
</div>
`;
